ompi/mpi/c/init_thread.c

```diff
@@ -48,9 +48,14 @@ int MPI_Init_thread(int *argc, char ***argv, int required,
                     int *provided)
 {
     int err;
+    char *env;
 
     ompi_hook_base_mpi_init_thread_top(argc, argv, required, provided);
 
+    if (NULL != (env = getenv("OMPI_MPI_THREAD_LEVEL")))  {
+        required = atoi(env);
```

Comment:
Is this documented anywhere?  It would seem pretty important to document this behavior -- e.g., in the man page.

Reply:
I also don't quite understand the commit message.  It says "Allow mpi_init_thread to override the MPI_THREAD_LEVEL", but it looks more like the commit allows an env variable to override the thread level set by the app in the call to MPI_INIT_THREAD.

This is new behavior that has not existed in 4.0.x and 4.1.x.  There are small (but nonzero) affects to backwards compatibility here, too.

Reply:
It's making it consistent with ompi_mpi_init, which also uses this ENV variable.

Reply:
`ompi_mpi_init()` does not check the environment variable for the thread level: `MPI_Init()` does.

We created an env variable "back door" for legacy apps that want to run with MPI threading support but couldn't change their source code to call `MPI_Init_thread()`.  Honestly, it was mainly a way for us to enable `MPI_THREAD_MULTIPLE` with all the existing OMPI MPI test codes without changing all of them to `MPI_Init_thread()`.

We intentionally did *not* put in the same backdoor to `MPI_Init_thread()` because `MPI_Init_thread()` allows the app to say exactly what it wants (where `MPI_Init()` does not).  Having an environment variable override to something that was specified as a parameter to `MPI_Init_thread()` seems... sketchy.